### DISCUSSION TITLE
springboot+druid+mybatis与kyuubi如何优雅集成

### DISCUSSION THREAD

**Discussion by liuyi-6161** [apache/kyuubi#3940] [General]
1.目前业务有springboot定时批处理，想利用user共享级别的spark engine来处理.可能是8-10点执行，其它时间回收engine
2.测试
 <dependency>
            <groupId>com.alibaba</groupId>
            <artifactId>druid-spring-boot-starter</artifactId>
            <version>1.1.21</version>
        </dependency>

        <dependency>
            <groupId>org.mybatis.spring.boot</groupId>
            <artifactId>mybatis-spring-boot-starter</artifactId>
            <version>2.2.1</version>
        </dependency>

        <dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>2.3.7</version>
            <exclusions>
                <exclusion>
                    <groupId>org.slf4j</groupId>
                    <artifactId>slf4j-api</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-core</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-classic</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

3.问题
经常出现 org.apache.kyuubi.KyuubiSQLException: Invalid SessionHandle ，感觉是会话机制的问题。

**Comment by pan3793** [comment]
check connection health before executing queries

**Reply by liuyi-6161** [reply]
如果进行检查，空闲的时候还是会起一个engine，目前是想做一个冷启动，使用的时候才起egine。spark动态分配+集群本身的自动扩缩容能力，尽可能提高资源利用率。

**Reply by pan3793** [reply]
I mean do check **BEFORE** execution, not **PERIODICALLY**
